skglm/experimental/sqrt_lasso.py

```diff
@@ -193,12 +200,18 @@ def path(self, X, y, alphas=None, eps=1e-3, n_alphas=10):
 
             l1_penalty.alpha = alphas[i]
             # no warm start for the first alpha
-            coef_init = coefs[i].copy() if i else np.zeros(n_features)
+            coef_init = coefs[i].copy() if i else np.zeros(n_features
+                                                           + self.fit_intercept)
 
             try:
-                coef, _, _ = self.solver_.solve(
-                    X, y, sqrt_quadratic, l1_penalty,
-                    w_init=coef_init, Xw_init=X @ coef_init)
+                if self.fit_intercept:
```

Comment:
There's duplicated here, I'd just put the definition of  Xw_init in the if else, but then coef, _, _ = ... can be factorized outside